ecs-agent/api/ecs/client/ecs_client.go

```diff
@@ -710,8 +711,21 @@ func (client *ecsClient) DiscoverServiceConnectEndpoint(containerInstanceArn str
 	return aws.StringValue(resp.ServiceConnectEndpoint), nil
 }
 
-func (client *ecsClient) discoverPollEndpoint(containerInstanceArn string) (*ecsmodel.DiscoverPollEndpointOutput,
+func (client *ecsClient) DiscoverSystemLogsEndpoint(containerInstanceArn string, availabilityZone string) (string,
```

Comment:
Why can't we get this information from the existing DPE call by adding the AZ into it? I imagine that's something every implementer would have

Reply:
If I'm understanding your question correctly-- this is just following the pattern of how we break down the DPE API response.

The `DiscoverPollEndpoint` function is for ACS specifically.

Reply:
Gotcha. I understand that DPE calls can be dangerous to add since we introduce risk of throttling, so I was wondering if we could avoid that by re-using a cached response.

It seems like no, since ACS connection happens after we start the agent

Reply:
We do use a cached response actually, if it's available and not expired. (See the `discoverPollEndpoint` function that each of the exported DPE functions invoke.)

Reply:
Have we verified the cache hit will not trigger for the new response which contains the logging endpoint?

Reply:
You bring up a good point that we don't want to assume that DiscoverSystemLogsEndpoint will always be the first invocation of the DPE API.

In the case that DPE is invoked without an AZ first, there could be a cache hit but with the system logs endpoint missing from it. I've made a change to check for the system logs endpoint if the AZ was provided instead of simply using the cache hit at all times.